sat/cli/bootsys/mgmt_power.py

```diff
@@ -255,37 +256,65 @@ def do_mgmt_shutdown_power(ssh_client, username, password, excluded_ncns, ncn_sh
-    # TODO: Is there a better way to get the hostname of the first master node?
     try:
-        other_ncns_by_role = get_and_verify_ncn_groups(excluded_ncns.union({'ncn-m001'}))
+        other_ncns_by_role = get_and_verify_ncn_groups(excluded_ncns)
     except FatalBootsysError as err:
         LOGGER.error(f'Not proceeding with NCN power off: {err}')
         raise SystemExit(1)
 
-    other_ncns = list({ncn for role in ('managers', 'storage', 'workers')
-                       for ncn in other_ncns_by_role[role]})
-
+    # Shutdown workers
+    if 'workers' in other_ncns_by_role:
+        worker_ncns = other_ncns_by_role['workers']
+        LOGGER.info(f'Shutting down worker NCNs: {", ".join(worker_ncns)}')
+        start_shutdown(worker_ncns, ssh_client)
+        finish_shutdown(worker_ncns, username, password, ncn_shutdown_timeout, ipmi_timeout)
+
+    # Shutdown masters (except ncn-m001)
+    if 'managers' in other_ncns_by_role:
+        master_ncns = other_ncns_by_role['managers']
+        master_ncns.remove('ncn-m001')  # Exclude ncn-m001
+        LOGGER.info(f'Shutting down master NCNs: {", ".join(master_ncns)}')
+        start_shutdown(master_ncns, ssh_client)
+        finish_shutdown(master_ncns, username, password, ncn_shutdown_timeout, ipmi_timeout)
+
+    # Unmap RBD on ncn-m001
+    m001_ncn = ['ncn-m001']
+    LOGGER.info(f'Unmapping RBD on ncn-m001: {", ".join(m001_ncn)}')
     try:
-        with IPMIConsoleLogger(other_ncns, username, password):
-            LOGGER.info(f'Sending shutdown command to other NCNs: {", ".join(other_ncns)}')
-            start_shutdown(other_ncns, ssh_client)
-            LOGGER.info(f'Waiting up to {ncn_shutdown_timeout} seconds for other NCNs to '
-                        f'reach powered off state according to ipmitool: {", ".join(other_ncns)}.')
-            finish_shutdown(other_ncns, username, password,
-                            ncn_shutdown_timeout, ipmi_timeout)
-            LOGGER.info('Shutdown and power off of all other NCNs complete.')
-    except ConsoleLoggingError as err:
-        LOGGER.error(f'Aborting shutdown of NCNs due failure to set up NCN console logging: {err}')
-        raise SystemExit(1)
+        # Execute rbdmap unmap-all command on ncn-m001
+        command = 'rbdmap unmap-all'
+        ssh_client.connect('ncn-m001')
+        stdin, stdout, stderr = ssh_client.exec_command(command)
+        exit_status = stdout.channel.recv_exit_status()
+        if exit_status != 0:
+            LOGGER.error(f'Failed to unmap RBD on ncn-m001. Command returned non-zero exit status: {exit_status}')
+            sys.exit(1)
+        else:
+            LOGGER.info('RBD unmapped successfully on ncn-m001')
+    except (paramiko.SSHException, socket.error) as err:
+        LOGGER.error(f'Error occurred while connecting to ncn-m001 or executing command: {err}')
+        sys.exit(1)
+
+    # Freeze Ceph on storage nodes and then shutdown
+    if 'storage' in other_ncns_by_role:
```

Comment:
with the else, would you want to have an exit , with shutdown failing

Reply:
No, I think it is okay for the `else` to just log a message that there are no storage nodes to shutdown because they are all excluded. The `get_mgmt_ncn_groups` function will already check that it was at least able to find some nodes of each management subrole (master, worker, storage) before we get to this point, so the only way these lists will be empty is if the nodes are requested to be excluded by the command-line arguments.

Reply:
thanks for clarifying Ryan, added it

Reply:
Yep, that looks good